tests/functional/test_rose_stem.py

```diff
@@ -211,57 +217,42 @@ def rose_stem_run_template(setup_stem_repo, pytestconfig):
-        # Run rose-stem
-        run_stem = subprocess.run(
-            split(rose_stem_cmd), capture_output=True,
-            cwd=setup_stem_repo['workingcopy']
-        )
-
-        # To assist with debugging fail horribly if the subproc'd rose-stem
-        # command returns non-zero:
-        if run_stem.returncode != 0:
-            if verbosity > 1:
-                # If -v print error details
-                print('\n\t'.join(
-                    [x.decode() for x in run_stem.stderr.split(b'\n')]
-                ))
-            if verbosity > 2:
-                # If -vv print replication instructions.
-                msg = (
-                    'To reproduce failure outside test environment:'
-                    f'\n\tcd {setup_stem_repo["workingcopy"]}'
-                    f'\n\texport FCM_CONF_PATH={os.environ["FCM_CONF_PATH"]}'
-                    f'\n\t{rose_stem_cmd}'
-                )
-                print(msg)
-                # If you want to debug add a breakpoint here:
-            msg = (
-                f'rose-stem command:\n {rose_stem_cmd} failed with'
-                f':\n{run_stem.stderr.decode()}'
-            )
-            raise SubprocessesError(msg)
-        outputpath = (
-            Path(setup_stem_repo['suite_install_dir']) /
-            'runN/opt/rose-suite-cylc-install.conf'
-        )
-        output = outputpath.read_text()
+        monkeymodule.setattr('sys.argv', ['stem'])
+        monkeymodule.chdir(setup_stem_repo['workingcopy'])
+        parser, opts = _get_rose_stem_opts()
+        [setattr(opts, key, val) for key, val in rose_stem_cmd.items()]
+
+        run_stem = SimpleNamespace()
+        run_stem.stdout = ''
+        try:
+            rose_stem(parser, opts)
+            run_stem.returncode = 0
+            run_stem.stderr = ''
+        except Exception as exc:
+            run_stem.returncode = 0
+            run_stem.stderr = exc
 
         return {
             'run_stem': run_stem,
-            'jobout_content': output,
+            'jobout_content': (
+                Path(setup_stem_repo['suite_install_dir']) /
+                'runN/opt/rose-suite-cylc-install.conf'
+            ).read_text(),
             **setup_stem_repo
         }
+
     yield _inner_fn
 
 
 @pytest.fixture(scope='class')
 def rose_stem_run_basic(rose_stem_run_template, setup_stem_repo):
-    rose_stem_cmd = (
-        "rose stem --group=earl_grey --task=milk,sugar --group=spoon,cup,milk "
-        f"--source={setup_stem_repo['workingcopy']} "
-        "--source=\"fcm:foo.x_tr\"@head "
-        f"--workflow-name {setup_stem_repo['suitename']}"
-    )
+    rose_stem_cmd = {
+        'stem_groups': ['earl_grey', 'milk,sugar', 'spoon,cup,milk'],
+        'stem_sources': [
+            str(setup_stem_repo['workingcopy']), "fcm:foo.x_tr@head"
+        ],
+        'workflow_name': setup_stem_repo['suitename']
+    }
     yield rose_stem_run_template(rose_stem_cmd)
```

Comment:
This no longer looks like a command, how about renaming the variables and argument to `rose_stem_opts` or something?